### PR TITLE
feat(ssz): Remove error from `HashTreeRoot()`

### PR DESCRIPTION
**Pull request by itsdevbear**
## Summary by CodeRabbit

- **New Features**
	- Simplified method signatures across various structures and interfaces by removing error return types from `HashTreeRoot` methods, enhancing usability and readability.
	- Streamlined error handling in multiple components, making code more concise and reducing verbosity.
  
- **Bug Fixes**
	- Improved error management in the state processing logic, potentially enhancing reliability during execution.

- **Documentation**
	- Minor updates to inline comments and documentation reflecting changes in method return types and error handling strategies.

- **Tests**
	- Refactored test cases to align with the new method signatures, ensuring that error handling is consistent with the updated implementations.

### CODE REVIEW

**Review by calbera — APPROVED** [review]
lgtm

**Review by nidhi-singh02 — APPROVED** [review]

File: mod/primitives/pkg/bytes/b20.go [hunk 1]
```diff
@@ -83,8 +83,8 @@ func (h B20) MarshalSSZ() ([]byte, error) {
 }
 
 // HashTreeRoot returns the hash tree root of the B20.
-func (h B20) HashTreeRoot() (B32, error) {
+func (h B20) HashTreeRoot() B32 {
 	var result [32]byte
 	copy(result[:], h[:])
-	return result, nil
+	return result
```
Comment (calbera) [inline mod/primitives/pkg/bytes/b20.go:89]: this and the b8 one could just be `return ToBytes32(h[:])`

File: mod/consensus-types/pkg/types/header_test.go [hunk 2]
```diff
@@ -77,8 +77,7 @@ func TestBeaconBlockHeader_HashTreeRoot(t *testing.T) {
 		common.Root{},
 	)
 
-	_, err := header.HashTreeRoot()
-	require.NoError(t, err)
+	_ = header.HashTreeRoot()
```
Comment (ocnc) [inline mod/consensus-types/pkg/types/header_test.go:80]:
```suggestion
	require.NotPanics(t, func() {
		_ = fork.HashTreeRoot()_ = header.HashTreeRoot()
	})
	
```

**Review by ocnc — APPROVED** [review]
looks same same

**Pull request merged**